riocli/deployment/update.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2023 Rapyuta Robotics
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import re
+import sys
+
+import click
+from click_help_colors import HelpColorsCommand
+
+from rapyuta_io import Client, DeploymentPhaseConstants
+
+from riocli.config import new_client
+from riocli.constants import Symbols, Colors
+from riocli.utils import tabulate_data
+from riocli.utils.spinner import with_spinner
+
+@click.command(
+    'update',
+    cls=HelpColorsCommand,
+    help_headers_color=Colors.YELLOW,
+    help_options_color=Colors.GREEN,
+)
+@click.option('--force', '-f', '--silent', is_flag=True, default=False,
+              help='Skip confirmation')
+@click.option('--update-all', '-a', is_flag=True, default=False,
+              help='Updates all deployments')
+@click.argument('deployment-name-or-regex', type=str, default="")
+@with_spinner(text="Updating deployment...")
+def update_deployment(force: bool, deployment_name_or_regex: str, update_all: bool, spinner=None) -> None:
+    """
+    Delete the deployment from the Platform
+    """
+    client = new_client()
+
+    if not(deployment_name_or_regex or update_all):
+        return
+
+    try:
+        deployments = fetch_deployments(client, deployment_name_or_regex, update_all)
+    except Exception as e:
+        spinner.text = click.style(
+            'Failed to update deployment(s): {}'.format(e), Colors.RED)
+        spinner.red.fail(Symbols.ERROR)
+        raise SystemExit(1) from e
+
+    if not deployments:
+        return
+
+    headers = ["name", "guid"]
+    data = [[deployment.name, deployment.deploymentId] for deployment in deployments]
```

Comment:
I think it would be good to show the current `status` or `phase` of the deployment for better clarity before confirming the operation.